Codeinwp/otter-blocks
# Add RankMath for Posts block

Closes #1871


### Summary


Add RankMath integration for Posts Block

Some notes on development:
- At first I tried to add inside the Edit component on the idea that each Block when is created its adding its own hook. The process was pretty messing with reactivity so I decided add a simple script that initiate at loading and scan everything at once -- reliable solution decoupled from react component, much easier to maintain and possible crashes on the block.
- I wanted to initiate the hook after the `rank-math-app.js` is loaded in the page via script loading. My attempt was unsuccessful so I used a simple try system.

### Screenshots 

https://github.com/Codeinwp/otter-blocks/assets/17597852/90e629e6-eff4-43da-bfec-9a28468f1289

----

### Test instructions


ℹ️ You will need RankMath plugin

1. Insert a Posts Block
2. Check the RankMath of the page.
3. You should not see the error message with `Add internal links in your content` since Post block will add references to other posts.

![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/5e544549-8eca-453a-a2ba-3c9c57ba7391)




---- 

### Checklist before the final review
